### PR TITLE
Update mysql.js to add OrderBy Rand()

### PR DESCRIPTION
**Pull request by yagobski**
Possibility to handle ORDER BY RAND()

### CODE REVIEW

**Comment by raymondfeng** [comment]
Please sign CLA - https://cla.strongloop.com/agreements/strongloop/loopback-connector-mysql

**Comment by raymondfeng** [comment]
If we decide to support `rand` as a special order, we need to fix all connectors.

**Comment by yagobski** [comment]
I don't undertand why you want to fix all connectors. This rand special order can be helpfull for Mysql connector.

**Comment by yagobski** [comment]
Can you merge this please

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

File: lib/mysql.js [hunk 1]
```diff
@@ -519,16 +519,20 @@ MySQL.prototype._buildWhere = function (model, conds) {
...
+    if (typeof order === 'string') {
+      order = [order];
+    }
+    return 'ORDER BY ' + order.map(function (o) {
+      var t = o.split(/[\s,]+/);
+      if (t.length === 1) {
+        return self.columnEscaped(model, o);
+      }
+      return self.columnEscaped(model, t[0]) + ' ' + t[1];
+    }).join(', ');
```
Comment (rlloyd2001) [inline lib/mysql.js]: This code has some styling issues that should be updated before merge.

**Comment by superkhau** [comment]
Can you add some tests to verify your changes and prevent regressions in the future?

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by loay** [comment]
@slnode test please

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by bbito** [comment]
The entire `buildOrderBy` method was removed with be54c1a on May 13, 2015 "Refactor the code to use base SqlConnector"

**Comment by loay** [comment]
@bbito yeah thanks. I saw that. I will either use it in `loopback-connector` or override it here.

**Comment by jannyHou** [comment]
@yagobski thank you for the contribution, any plan on @loay's suggestion? see https://github.com/strongloop/loopback-connector-mysql/pull/62#issuecomment-286514761

**Comment by darknos** [comment]
+1

I'd refactor it to allow use functions in order class. It is useful not only with `rand()` but with any complex ordering or using function like `getdistance()` when it is not possible to get all records in memory to find N first records.

**Comment by loay** [comment]
closing this pr, since the function is being moved to `loopback-connector`

**Pull request closed without merging**